packages/spidermonkey.rb

```diff
@@ -0,0 +1,63 @@
+require 'package'
+
+class Spidermonkey < Package
+  description 'SpiderMonkey is Mozilla\'s JavaScript engine written in C and C++.'
+  homepage 'https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey'
+  version '84.0'
+  compatibility 'all'
+  source_url 'https://archive.mozilla.org/pub/firefox/releases/84.0/source/firefox-84.0.source.tar.xz'
+  source_sha256 '23273ef0165b243f5d0908c38e7854d38070282c9b526e8d93b7503cd5f69138'
+
+  depends_on 'yasm' # Might be a build dependency, undergoing testing
+  depends_on 'rust' => :build
+  
+  def self.prebuild
+    puts
+    puts "Building spidermonkey requires at least 6.4 GB of space available on /mnt/stateful_partition.".lightblue
+    puts "It WILL fail without this space. Press Ctrl + C to cancel.".lightblue
+    puts
+    sleep(5)
+  end
+
+  case ARCH
+  when 'aarch64', 'armv7l'
+    CREW_BUILD = 'armv7l-cros-linux-gnueabihf'
+  when 'i686'
+    CREW_BUILD = 'i686-cros-linux-gnu'
+  when 'x86_64'
+    CREW_BUILD = 'x86_64-cros-linux-gnu'
+  end
+
+  def self.build
+    ENV['LC_ALL'] = 'C' # https://bugs.gentoo.org/746215
+    ENV['MACH_USE_SYSTEM_PYTHON'] = "#{CREW_PREFIX}/bin/python3"
+    ENV['CC'] = 'gcc' # Defaults to clang, but configure.py complains about oudated libstdc++ when using clang
+    ENV['CXX'] = 'g++'
+    ENV['AR'] = 'ar'
+    ENV['AS'] = 'as'
+    ENV['RANLIB'] = 'ranlib'
+    ENV['NM'] = 'nm'
+    Dir.chdir 'build' do
+      system "#{CREW_PREFIX}/bin/python3 \
+            ../configure.py --prefix=#{CREW_DEST_PREFIX} \
+            --libdir=#{CREW_DEST_LIB_PREFIX} \
+            --host=#{CREW_BUILD } \
```

Comment:
Extra space?
`#{CREW_BUILD }`

Reply:
I literally hate my IDE, i wish chromeos had a better one

Reply:
What is your IDE?  Try codium if you haven't already.

Reply:
https://chrome.google.com/webstore/detail/code-pad-text-editor/adaepfiocmagdimjecpifghcgfjlfmkh
this is my IDE, just make sure to go into the extension settings and disable its access to https://google-analytics.com

And does codium work on a chromebook?

Reply:
Try `crew install codium` and find out.